sim/src/generic_eps_hardware_model.cpp

```diff
@@ -19,49 +20,128 @@ namespace Nos3
 
         /* Get on a protocol bus */
         /* Note: Initialized defaults in case value not found in config file */
-        std::string bus_name = "usart_29";
-        int node_port = 29;
+        std::string bus_name = "i2c_0";
+        int bus_address = 0x2B;
         if (config.get_child_optional("simulator.hardware-model.connections")) 
         {
             /* Loop through the connections for hardware model */
             BOOST_FOREACH(const boost::property_tree::ptree::value_type &v, config.get_child("simulator.hardware-model.connections"))
             {
                 /* v.second is the child tree (v.first is the name of the child) */
-                if (v.second.get("type", "").compare("usart") == 0)
+                if (v.second.get("type", "").compare("i2c") == 0)
                 {
                     /* Configuration found */
                     bus_name = v.second.get("bus-name", bus_name);
-                    node_port = v.second.get("node-port", node_port);
+                    bus_address = v.second.get("bus-address", bus_address);
                     break;
                 }
             }
         }
-        _uart_connection.reset(new NosEngine::Uart::Uart(_hub, config.get("simulator.name", "generic_eps_sim"), connection_string, bus_name));
-        _uart_connection->open(node_port);
-        sim_logger->info("Generic_epsHardwareModel::Generic_epsHardwareModel:  Now on UART bus name %s, port %d.", bus_name.c_str(), node_port);
-    
-        /* Configure protocol callback */
-        _uart_connection->set_read_callback(std::bind(&Generic_epsHardwareModel::uart_read_callback, this, std::placeholders::_1, std::placeholders::_2));
-
-        /* Get on the command bus*/
-        std::string time_bus_name = "command";
+        _i2c_slave_connection = new I2CSlaveConnection(this, bus_address, connection_string, bus_name);
+        sim_logger->info("Generic_epsHardwareModel::Generic_epsHardwareModel:  Now on I2C bus name %s as address 0x%02x.", bus_name.c_str(), bus_address);
+
+        /* Get on the command bus - EPS receives commands here */
+        _time_bus_name = "command";
```

Comment:
Change _time_bus_name to _command_bus_name and declare it as std::string here (removing it from the class definition... it is never used outside this constructor).